package.json

```diff
@@ -16,38 +16,91 @@
   },
   "devDependencies": {
     "@aws-cdk/aws-apprunner-alpha": "2.20.0-alpha.0",
-    "@types/archiver": "^5.3.1",
-    "@types/chai": "^4.3.1",
-    "@types/mocha": "^9.1.0",
-    "@types/node": "^14.17.0",
-    "@typescript-eslint/eslint-plugin": "^5.19.0",
-    "@typescript-eslint/parser": "^5.19.0",
-    "chai": "^4.3.6",
-    "eslint": "^8.13.0",
-    "eslint-config-prettier": "^8.5.0",
-    "mocha": "^9.2.2",
+    "@pulumi/aws": "^6.32.0",
+    "@pulumi/aws-native": "^0.108.0",
+    "@pulumi/docker": "^4.5.0",
+    "@pulumi/pulumi": "^3.117.0",
+    "@types/archiver": "^6.0.2",
+    "@types/jest": "^29.5.2",
+    "@types/node": "^20.12.13",
+    "@typescript-eslint/eslint-plugin": "^7.11.0",
+    "@typescript-eslint/parser": "^7.11.0",
+    "aws-cdk-lib": "^2.20.0",
+    "constructs": "^10.0.111",
+    "eslint": "^8",
+    "eslint-config-prettier": "^9.1.0",
+    "jest": "^29.5.0",
+    "jest-junit": "^15",
     "prettier": "^2.6.2",
-    "ts-node": "^10.7.0",
-    "typescript": "^4.6.3"
+    "ts-jest": "^29.1.0",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  },
+  "peerDependencies": {
+    "@pulumi/aws": "^6.32.0",
```

Comment:
I'm curious why we deviate from what we do in other providers. E.g. awsx has them as direct dependencies: https://github.com/pulumi/pulumi-awsx/blob/master/sdk/nodejs/package.json#L17

From my understanding peerDependencies would be more apt for plugins, but this here is rather a library, right?

Reply:
I can switch the `@pulumi/*` dependencies back to direct dependencies if we want. The CDK dependencies definitely need to be `peerDependencies`. `peerDependencies` allow us to specify a minimum required version that works with our library and then the user can decide which version to install. I think as we (or the community) develops more component libraries it might be more important to use `peerDependencies`.

https://dev.to/aws-builders/correctly-defining-dependencies-in-l3-cdk-constructs-45p